### PR TITLE
[libcxx] don't `#include <cwchar>` if wide chars aren't enabled

### PR DESCRIPTION
**Pull request by stuij**
Pull request #96032 unconditionall adds the `cwchar` include in the `format` umbrella header. However support for wchar_t can be disabled in the build system (LIBCXX_ENABLE_WIDE_CHARACTERS).

This patch guards against inclusion of `cwchar` in `format` by checking the `_LIBCPP_HAS_NO_WIDE_CHARACTERS` define.

For clarity I've also merged the include header section that `cwchar` was in with the one above as they were both guarded by the same `#if` logic.

### CODE REVIEW

File: libcxx/include/format [hunk 1]
```diff
@@ -237,21 +237,21 @@ namespace std {
...
 #  include <optional>
+#  include <queue>
+#  include <stack>
 #  include <stdexcept>
 #  include <string>
 #  include <string_view>
 #  include <tuple>
-#endif
 
-#if !defined(_LIBCPP_REMOVE_TRANSITIVE_INCLUDES) && _LIBCPP_STD_VER <= 20
```
Comment (Stylie777) [inline libcxx/include/format:251]: Why are we changing the behaviour here? This PR is protecting the include for `cwchar`, I think this needs to be done in a separate commit/PR if this is not related. If it is, can there be an explanation as to why?

File: libcxx/include/format [hunk 2]
```diff
@@ -237,21 +237,21 @@ namespace std {
 #  include <cstdint>
 #  include <cstdlib>
 #  include <cstring>
-#  include <cwchar>
 #  include <initializer_list>
 #  include <limits>
+#  include <locale>
```
Comment (statham-arm) [inline libcxx/include/format:242]: The commit message doesn't say anything about why you've _also_ made the includes of `<locale>`, `<queue>` and `<stack>` unconditional. Was that change included in this patch by mistake? If it was on purpose, could you update the commit message to explain why it's related or necessary?

Comment (stuij) [reply libcxx/include/format:242]: We're not making them unconditional. If you expand this window to see the lines above, you'll see that the exact same conditional guards the includes above.

I merged the two sections to make the code clearer.

File: libcxx/include/format (hunk 1 above)
Comment (stuij) [reply libcxx/include/format:251]: We're not changing the behaviour. If you expand this window to see the lines above, you'll see that the exact same conditional guards the includes above.

I merged the two sections to make the code clearer.

Comment (Stylie777) [reply libcxx/include/format:251]: Can that be made clearer in the commit message?

File: libcxx/include/format (hunk 2 above)
Comment (stuij) [reply libcxx/include/format:242]: As I think we'll loose or bury this convo if I force-push, I've updated the description in my automated first comment in this thread. I'll update it when I merge it. Unless there's a better way to handle this.

File: libcxx/include/format (hunk 1 above)
Comment (stuij) [reply libcxx/include/format:251]: As I think we'll loose or bury this convo if I force-push, I've updated the description in my automated first comment in this thread. I'll update the description when I merge it. Unless there's a better way to handle this.

Comment (statham-arm) [reply libcxx/include/format:251]: I have no idea whether that will do what you want, so good luck with getting the right version of the commit message into the final commit!

**Review by statham-arm — APPROVED** [review]

File: libcxx/include/format (hunk 1 above)
Comment (Stylie777) [reply libcxx/include/format:251]: If you force-push over the current commit it should keep this conversation, not sure GitHub will delete it.

Comment (stuij) [reply libcxx/include/format:251]: ok, let me give it a shot now that I've got approval.

Comment (stuij) [reply libcxx/include/format:251]: Ah that did work. Googling says comments don't disappear if the lines they're on don't change. Seems like they do still disappear if the lines change: https://github.com/orgs/community/discussions/3478.

**Comment by stuij** [comment]
@philnik777 ok, sorry for that!

**Comment by ldionne** [comment]
I think this should be in LLVM 19: this is a regression from LLVM 18
